client/chain/chain.go

```diff
@@ -180,12 +180,25 @@ type chainClient struct {
 	canSign bool
 }
 
-// NewChainClient creates a new gRPC client that communicates with gRPC server at protoAddr.
-// protoAddr must be in form "tcp://127.0.0.1:8080" or "unix:///tmp/test.sock", protocol is required.
+// Deprecated: Use NewChainClientWithMarketsAssistant instead.
 func NewChainClient(
 	ctx client.Context,
 	network common.Network,
 	options ...common.ClientOption,
+) (ChainClient, error) {
+	assistant, err := core.NewMarketsAssistant(network.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewChainClientWithMarketsAssistant(ctx, network, assistant, options...)
+}
+
+func NewChainClientWithMarketsAssistant(
+	ctx client.Context,
+	network common.Network,
+	marketsAssistant core.MarketsAssistant,
+	options ...common.ClientOption,
```

Comment:
I understand your point. I thought of using a config function, but with a config function I did not use it because:
- I want to make the MarketAssistant parameter mandatory
- I wanted to create a new instance creation function to be able to explicitly mark the old one as deprecated

Do you think we should ignore those two points?